dali/test/python/test_pool.py

```diff
@@ -73,93 +78,65 @@ def wrapper():
-# of calculating the second batch, just in case we wait a few seconds
-@check_pool
-def test_pool_overwrite_single_batch(start_method):
-    callbacks = [simple_callback]
-    pool = create_pool(callbacks, queue_depth=1, num_workers=1, start_method=start_method)
-    pids = get_pids(pool)
-    pid = pids[0]
-    tasks_0 = [(SampleInfo(0, 0, 0),)]
-    tasks_1 = [(SampleInfo(1, 0, 1),)]
-    pool.schedule_batch(context_i=0, batch_i=0, dst_chunk_i=0, tasks=tasks_0)
-    pool.schedule_batch(context_i=0, batch_i=1, dst_chunk_i=0, tasks=tasks_1)
-    time.sleep(5)
-    batch_0 = pool.receive_batch(context_i=0)
-    batch_1 = pool.receive_batch(context_i=0)
-    for task, sample in zip(tasks_1, batch_0):
-        np.testing.assert_array_equal(answer(pid, *task), sample)
-    for task, sample in zip(tasks_1, batch_1):
-        np.testing.assert_array_equal(answer(pid, *task), sample)
-    pool.close()
-
-
-# Test that with bigger queue depth we will still overwrite the memory used as the results
-@check_pool
-def test_pool_overwrite_multiple_batch(start_method):
-    callbacks = [simple_callback]
-    pool = create_pool(callbacks, queue_depth=3, num_workers=1, start_method=start_method)
-    pids = get_pids(pool)
-    pid = pids[0]
-    tasks_list = [(i, [(SampleInfo(i, 0, i),)]) for i in range(4)]
-    for i, tasks in tasks_list:
-        pool.schedule_batch(context_i=0, batch_i=i, dst_chunk_i=i%3, tasks=tasks)
-    batches = [pool.receive_batch(context_i=0) for i in range(4)]
-    tasks_batches = zip(tasks_list, batches)
-    _, tasks_3 = tasks_list[3]
-    for (i, tasks), batch in tasks_batches:
-        if i == 0:
-            tasks_to_compare = tasks_3
-        else:
-            tasks_to_compare = tasks
-        for task, sample in zip(tasks_to_compare, batch):
+    with create_pool(callbacks, queue_depth=1, num_workers=1, start_method=start_method) as pool:
+        pids = get_pids(pool)
+        pid = pids[0]
+        tasks = [(SampleInfo(i, i, 0),) for i in range(10)]
+        pool.schedule_batch(context_i=0, batch_i=0, dst_chunk_i=0, tasks=tasks)
+        batch = pool.receive_batch(context_i=0)
+        for task, sample in zip(tasks, batch):
             np.testing.assert_array_equal(answer(pid, *task), sample)
-    pool.close()
 
 
-# Test that we can hold as many results as the queue depth
 @check_pool
-def test_pool_no_overwrite_batch(start_method):
+def test_pool_scheduled_tasks_cleanup(start_method):
     callbacks = [simple_callback]
-    for depth in [1, 2, 4, 8]:
-        pool = create_pool(callbacks, queue_depth=depth, num_workers=1, start_method=start_method)
+    with create_pool(callbacks, queue_depth=3, num_workers=1, start_method=start_method) as pool:
         pids = get_pids(pool)
         pid = pids[0]
-        tasks_list = [(i, [(SampleInfo(i, 0, i),)]) for i in range(depth)]
+        tasks_list = [(i, [(SampleInfo(i, 0, i),)]) for i in range(3)]
```

Comment:
Why `range(3)`? How that relates to `queue_depth=3`?

Reply:
Yes, it's the number of batches that can fit into shm circular buffer for data from workers.

Reply:
Maybe you can bring back a variable that sets this? Now this 3 looks quite magic and I'm not sure where the `3` value is on purpose and where just like that.

Reply:
That's good point, I added it back (and actually testing on a few different depths).